test/gen_source.rb

```diff
@@ -0,0 +1,113 @@
+require 'optparse'
+require 'yaml'
+require 'erb'
+
+
+REGEXT_PRETTY = %r{
+ =\s            # We are interested to the right of the equal sign
+ (
+    ""|         # Empty string
+    ".*?[^\\]"| # String who can contain space and quoted string
+    [^\s,]+     # Anything except space and comma
+ )
+}x
+
+SOURCE_TEMPLATE =  <<-TEXT
+/* Code generaed by source_callbacks_generator.rb */
+
+#include "component.h"
+#include "create.h"
+
+void btx_push_usr_messages(struct common_data_s *common_data) {
+
+    <%- data.each do | entry | -%>
+    <%- entry.fetch(:times,1).times do -%>
+    btx_push_message_<%= entry[:name].gsub(":","_") %>(common_data,<%=  entry[:field_values].join(",") %>);
+    <%- end -%>
+    <%- end -%>
+}
+
+TEXT
+
+def parse_log(input_path)
+  data_list = []
+  File.open(input_path, "r") do |f|
+    f.each_line do |line|
+
+      # Line format support checks
+      match = line.match(REGEXT_PRETTY)
+      raise "Unsupported format for '#{line}'." unless match
+
+      i = line.index('{')
+      head, _, tail = line.partition(": {")
+      
+      data = {}
+      data[:name]=head.gsub(/[^0-9A-Za-z\-]/, '_') # Should reuse metababel mangling
+      data[:field_values] = tail.scan(REGEXT_PRETTY).flatten
+      data_list.append(data)
```

Comment:
please use `.map`

Reply:
This change kept me very thoughtful, If you mean to use f.map in place of f.each_line, each_line will load line by line into memory, but, for map, i am not sure how it did since it use enumerators its supposed to do the same but pulling batches of lines for map to work, not able to find how it is implemented.